Competition2017/src/competition/subsystems/vision/VisionSubsystem.java

```diff
@@ -0,0 +1,165 @@
+package competition.subsystems.vision;
+
+import org.apache.log4j.Logger;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+import competition.networking.NetworkedCommunicationServer;
+import competition.networking.OffboardCommPacket;
+import competition.networking.OffboardCommunicationServer;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+import json.JSONArray;
+import json.JSONObject;
+import xbot.common.command.BaseSubsystem;
+import xbot.common.command.PeriodicDataSource;
+import xbot.common.injection.wpi_factories.WPIFactory;
+import xbot.common.properties.DoubleProperty;
+import xbot.common.properties.XPropertyManager;
+import edu.wpi.first.wpilibj.Timer;
+
+@Singleton
+public class VisionSubsystem extends BaseSubsystem implements PeriodicDataSource {
+    private static Logger log = Logger.getLogger(VisionSubsystem.class);
+
+    OffboardCommunicationServer server;
+    List<DetectedLiftPeg> trackedLiftPegs = Collections.synchronizedList(new ArrayList<>());
+    List<DetectedBoiler> trackedBoilers = Collections.synchronizedList(new ArrayList<>());
+    
+    private static final String targetSnapshotPacketType = "trackedTargetsSnapshot";
+    private static final String trackedLiftPegsProperty = "trackedLiftPegs";
+    private static final String trackedBoilersProperty = "trackedBoilers";
+
+    private volatile double lastTimePacketRecieved = -1;
+    private volatile double lastPacketCounterResetTime = -1;
+    private volatile int numPacketsSinceReset = 0;
+
+    private volatile boolean isConnected = false;
+    
+    private DoubleProperty connectionTimeoutThreshold;
+    private DoubleProperty connectionReportInterval;
+
+    @Inject
+    public VisionSubsystem(WPIFactory factory, XPropertyManager propManager, OffboardCommunicationServer server) {
+        log.info("Creating VisionSubsystem");
```

Comment:
with the new logging settings Alex did, you don't need to include the name, so this can just be `log.info("Creating");`